src/lib/hooks/useTokenList/index.ts

```diff
@@ -40,25 +40,23 @@ export default function useTokenList(list?: string | TokenInfo[]): WrappedTokenI
       tokens.then(tokensToChainTokenMap).then(setChainTokenMap).catch(setError)
     }
   }, [chainId, library, list, setChainTokenMap])
-
-  return useMemo(() => {
-    return Object.values((chainId && chainTokenMap[chainId]) || {}).map(({ token }) => token)
-  }, [chainId, chainTokenMap])
 }
 
-export type TokenMap = { [address: string]: Token }
+export type TokenMap<T extends Token> = { [address: string]: T }
 
-export function useTokenMap(): TokenMap {
+export function useTokenMap(): TokenMap<WrappedTokenInfo> {
```

Comment:
Are you sure this doesn't break other typings? This also makes it less generic (because we're using our own interface instead of the one provided by the sdk), so I'd rather continue using `Token`.

Reply:
This... does not look familiar to me. Not sure where this change came from. Might have been an editor thing? idk. reverting.

Reply:
ah this was in the cherry-picked commit, will take a closer look at these changes